specs/interop.md

```diff
@@ -0,0 +1,188 @@
+- The chain id of the initiating message MUST be in the interop set
+- The relaying message MUST be initiated by an externally owned account such that the top level EVM call frame enters the `CrossL2Inbox`
+
+#### Only EOA Invariant
+
+The `onlyEOA` invariant on relaying a cross chain message enables static analysis on relaying messages.
+This allows for the derivation pipeline and block builders to reject relaying messages that do not
+have a corresponding initiating message.
+
+### L1Block
+
+The `L1Block` contract is updated to include the set of allowed chains. The L1 Atrributes transaction
+sets the set of allowed chains. The `L1Block` contract MUST provide a public getter to check is a particular
+chain is in the interop set.
+
+The `setL1BlockValuesInterop()` function MUST be called on every block after the interop upgrade block.
+The interop upgrade block itself MUST include a call to `setL1BlockValuesEcotone`.
+
+## SystemConfig
+
+The `SystemConfig` is updated to manage the interop set. The chain operator can add or remove chains from the interop set
+through the `SystemConfig`. A new `ConfigUpdate` event `UpdateType` enum is added that corresponds to a change in the interop set.
+
+The `SystemConfig` MUST enforce that the maximum size of the interop set is `type(uint8).max` or 255.
+
+### `INTEROP_SET` UpdateType
+
+When a `ConfigUpdate` event is emitted where the `UpdateType` is `INTEROP_SET`, the L2 network will update its interop set.
+The chain operator SHOULD be able to add or remove chains from the interop set.
+
+## L1Attributes
+
+The L1 Atrributes transaction is updated to include the interop set. Since the interop set is dynamically sized,
+a `uint8` "interopSetSize" parameter prefixes tightly packed `uint256` values that represent each chain id.
+
+| Input arg         | Type                    | Calldata bytes          | Segment |
+| ----------------- | ----------------------- | ----------------------- | --------|
+| {0x760ee04d}      | bytes4                  | 0-3                     | n/a     |
+| baseFeeScalar     | uint32                  | 4-7                     | 1       |
+| blobBaseFeeScalar | uint32                  | 8-11                    |         |
+| sequenceNumber    | uint64                  | 12-19                   |         |
+| l1BlockTimestamp  | uint64                  | 20-27                   |         |
+| l1BlockNumber     | uint64                  | 28-35                   |         |
+| basefee           | uint256                 | 36-67                   | 2       |
+| blobBaseFee       | uint256                 | 68-99                   | 3       |
+| l1BlockHash       | bytes32                 | 100-131                 | 4       |
+| batcherHash       | bytes32                 | 132-163                 | 5       |
+| interopSetSize    | uint8                   | 164-165                 | 6       |
```

Comment:
This would limit the number of chains in the interop set to 255. Is this enough?